File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ConfirmRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public abstract class ConfirmRule<T> extends Rule<T> {


    public ConfirmRule(Class type) {
        super(type);
    }

    public abstract boolean isValid(T value1, T value2);

    @Override
    public boolean isValid(T value) {
        return isValid(value, value);
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ConfirmRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

abstract class ConfirmRule<T>(type: Class<T>) : Rule<T>(type) {

    abstract fun isValid(value1: T?, value2: T?): Boolean

    override fun isValid(value: T?): Boolean {
        return isValid(value, value)
    }
}
